src/momento/momento_signer.py

```diff
@@ -0,0 +1,140 @@
+
+        Args:
+            jwk_json_string: the JSON string of the JWK key. This can be obtained from create_signing_key response.
+
+        Raises:
+            InvalidArgumentError: If the supplied private key is not valid.
+        """
+        try:
+            jwk_json: Dict[str, str] = json.loads(jwk_json_string)
+        except json.decoder.JSONDecodeError:
+            raise InvalidArgumentError(f"Invalid JWK Json String: {jwk_json_string}")
+
+        if "kid" not in jwk_json:
+            raise InvalidArgumentError(f"JWK missing kid attribute: {jwk_json_string}")
+
+        if "alg" not in jwk_json:
+            raise InvalidArgumentError(f"JWK missing alg attribute: {jwk_json_string}")
+
+        try:
+            self._jwk: PyJWK = PyJWK.from_dict(jwk_json)  # type: ignore[no-untyped-call, misc]
+        except jwt.exceptions.PyJWKError:
+            raise InvalidArgumentError(f"Invalid JWK: {jwk_json_string}")
+
+        self._kid = jwk_json["kid"]
+        self._alg = jwk_json["alg"]
+
+    def sign_access_token(self, signing_request: SigningRequest) -> str:
+        """Creates an access token to be used as a JWT token.
+
+        Args:
+            signing_request: Contains parameters for the generated token.
+
+        Returns:
+            str
+        """
+        claims: Dict[str, object] = {
+            "exp": signing_request.expiry_epoch_seconds(),
+            "cache": signing_request.cache_name(),
+            "key": signing_request.cache_key(),
+        }
+
+        if signing_request.cache_operation() == CacheOperation.GET:
+            claims["method"] = ["get"]
+        elif signing_request.cache_operation() == CacheOperation.SET:
+            claims["method"] = ["set"]
+            if signing_request.ttl_seconds() is not None:
+                claims["ttl"] = signing_request.ttl_seconds()
+        else:
+            raise NotImplementedError(
+                f"Unrecognized Operation: {signing_request.cache_operation()}"
+            )
+
+        # jwt.encode will automatically insert "typ" and "alg" into the header for us.
+        # We still need to specify "kid" to be included in the header however.
+        return jwt.encode(
+            claims, self._jwk.key, algorithm=self._alg, headers={"kid": self._kid}  # type: ignore[misc]
+        )
+
+    def create_presigned_url(
+        self, hostname: str, signing_request: SigningRequest
+    ) -> str:
+        """Creates a pre-signed HTTPS URL.
+
+        Args:
+            hostname: Hostname of SimpleCacheService. This value can be obtained from create_signing_key response.
+            signing_request: Contains parameters for the generated URL.
+
+        Returns:
+            str
+        """
+        token = self.sign_access_token(signing_request)
+        if signing_request.cache_operation() == CacheOperation.GET:
+            return f"https://{hostname}/cache/get/{signing_request.cache_name()}/{signing_request.cache_key()}?token={token}"
```

Comment:
Are you sure you don't want to url encode the cache key?
```
➜  cacheservice git:(envoydoc) ✗ momento cache set --name '.-_.-_.-' --key '///' --value 'you know' --ttl 90
[2022-04-07T23:29:05Z INFO  momento::commands::cache::cache_cli] setting key: /// into cache: .-_.-_.-
[2022-04-07T23:29:05Z INFO  momento::commands::cache::cache_cli] set success
```

This clearly makes no sense at all:
```
➜  cacheservice git:(envoydoc) ✗ curl https://cache.cell-alpha-dev.preprod.a.momentohq.com/cache/get/.-_.-_.-////\?token\=$JWS_ALPHA 
```

But this works:
```
curl https://cache.cell-alpha-dev.preprod.a.momentohq.com/cache/get/.-_.-_.-/%2f%2f%2f\?token\=$JWS_ALPHA
you know2
```

Reply:
Yess.... good catch. 

I think we have this bug in c# sdk as well. Ticket filed: https://github.com/momentohq/client-sdk-csharp/issues/43